### PR TITLE
Make iterators consistent with SDK pattern 

### PR DESCRIPTION
**Pull request by mpoke**
Closes https://github.com/cosmos/interchain-security/issues/449, https://github.com/cosmos/interchain-security/issues/446

### CODE REVIEW

File: x/ccv/provider/keeper/keeper_test.go [hunk 1]
```diff
@@ -369,28 +369,28 @@ func TestVscSendTimestamp(t *testing.T) {
...
 		i++
-		return true
+		return false // do not stop
 	})
 	require.Equal(t, 2, i)
 
 	// delete VSC send timestamps
 	var ids []uint64
-	providerKeeper.IterateVscSendTimestamps(ctx, testCases[0].chainID, func(vscID uint64, _ time.Time) bool {
+	providerKeeper.IterateVscSendTimestamps(ctx, testCases[0].chainID, func(vscID uint64, _ time.Time) (stop bool) {
```
Comment (smarshall-spitzbart) [inline x/ccv/provider/keeper/keeper_test.go:382]: Don't need to iterate here twice, but that's not important for this PR

File: x/ccv/provider/keeper/proposal.go [hunk 2]
```diff
@@ -159,14 +159,14 @@ func (k Keeper) StopConsumerChain(ctx sdk.Context, chainID string, lockUbd, clos
...
-		k.IterateOverUnbondingOpIndex(ctx, chainID, func(vscID uint64, ids []uint64) bool {
+		k.IterateOverUnbondingOpIndex(ctx, chainID, func(vscID uint64, ids []uint64) (stop bool) {
 			// iterate over the unbonding operations for the current VSC ID
 			var maturedIds []uint64
 			for _, id := range ids {
 				unbondingOp, found := k.GetUnbondingOp(ctx, id)
 				if !found {
 					err = fmt.Errorf("could not find UnbondingOp according to index - id: %d", id)
-					return false
+					return true // stop the iteration
```
Comment (smarshall-spitzbart) [inline x/ccv/provider/keeper/proposal.go:169]: Also not relevant for this PR necessarily, but we might want to panic here? And/or consider if we should keep iterating if we don't panic

File: x/ccv/provider/keeper/relay.go [hunk 3]
```diff
@@ -157,7 +157,7 @@ func (k Keeper) sendValidatorUpdates(ctx sdk.Context) {
 				// CCV channel not established:
 				// store the packet data to be sent once the CCV channel is established
 				k.AppendPendingVSCs(ctx, chainID, []ccv.ValidatorSetChangePacketData{packetData})
-				return true // go to next consumer chain
+				return false // go to next consumer chain
```
Comment (smarshall-spitzbart) [inline x/ccv/provider/keeper/relay.go:160]:
```suggestion
				return false // do not stop iteration, go to next consumer chain
```

**Review by smarshall-spitzbart — APPROVED** [review]
LGTM, thanks @mpoke! Just a couple small comments. Following those we can close https://github.com/cosmos/interchain-security/issues/446. 

https://github.com/cosmos/interchain-security/issues/454 will be useful to prove out correct behavior of iterators

File: x/ccv/provider/keeper/proposal.go (hunk 2 above)
Comment (mpoke) [reply x/ccv/provider/keeper/proposal.go:169]: We do return an error below, see https://github.com/cosmos/interchain-security/blob/f2e403a25f851678a0dd1b07fdaa7a8ca55abc38/x/ccv/provider/keeper/proposal.go#L195

File: x/ccv/provider/keeper/keeper_test.go (hunk 1 above)
Comment (smarshall-spitzbart) [reply x/ccv/provider/keeper/keeper_test.go:382]: The two calls to `IterateVscSendTimestamps` as they exist could be replaced with one iterator
```go
	var ids []uint64
	i = 0
	providerKeeper.IterateVscSendTimestamps(ctx, testCases[0].chainID, func(vscID uint64, ts time.Time) (stop bool) {
		ids = append(ids, vscID)
		require.Equal(t, vscID, testCases[i].vscID)
		require.Equal(t, ts, testCases[i].ts)
		i++
		return false // do not stop
	})
	require.Equal(t, 2, i)
	for _, vscID := range ids {
		providerKeeper.DeleteVscSendTimestamp(ctx, testCases[0].chainID, vscID)
	}
```

Comment (mpoke) [reply x/ccv/provider/keeper/keeper_test.go:382]: Oh, yeah. Not important though.

**Comment by danwt** [comment]
Whats the reason for this going into [marius/435-client-expired-consumer](https://github.com/cosmos/interchain-security/tree/marius/435-client-expired-consumer) and not main, aren't we trying to minimize diffs?

**Comment by mpoke** [comment]
> Whats the reason for this going into [marius/435-client-expired-consumer](https://github.com/cosmos/interchain-security/tree/marius/435-client-expired-consumer) and not main, aren't we trying to minimize diffs?

Once #435 is merged, this PR will be based on main. I basically try to minimized conflicts.

**Comment by danwt** [comment]
I checked and it doesn't seem that any iterators or callsites were missed or wrong 👍 Nice work

**Comment by MSalopek** [comment]
Ready for another review cycle after rebasing onto main

**Review by mpoke — COMMENTED** [review]
Thanks @MSalopek. LGTM. Cannot approve since I opened the PR.

**Review by smarshall-spitzbart — APPROVED** [review]
Looks great!

File: x/ccv/provider/keeper/keeper_test.go [hunk 4]
```diff
@@ -367,28 +368,174 @@ func TestVscSendTimestamp(t *testing.T) {
...
+	defer ctrl.Finish()
+
+	type chanToChain struct {
+		chainID   string
+		channelID string
+	}
+
+	cases := []chanToChain{
+		{
+			chainID:   "chain-1",
```
Comment (smarshall-spitzbart) [inline x/ccv/provider/keeper/keeper_test.go:447]: 🙌

File: x/ccv/consumer/keeper/genesis_test.go [hunk 5]
```diff
@@ -390,6 +390,6 @@ func assertHeightValsetUpdateIDs(t *testing.T, ctx sdk.Context, ck *consumerkeep
 		require.Equal(t, heighValsetUpdateIDs[ctr].Height, height)
```
Comment (sainoe) [inline x/ccv/consumer/keeper/genesis_test.go:390]: nit: can you also name the `bool` callback output as `stop` above (line 389)

**Review by sainoe — APPROVED** [review]
LGTM!

File: x/ccv/consumer/keeper/genesis_test.go (hunk 5 above)
Comment (MSalopek) [reply x/ccv/consumer/keeper/genesis_test.go:390]: Done

**Pull request merged**